src/inc/msquic.h

```diff
@@ -1605,6 +1605,62 @@ QUIC_STATUS
     _In_opt_ void* ClientSendContext
     );
 
+//
+// Connection Pool API
+//
+
+//
+// Closes the Connection Pool API table and cleans up the API reference.
+//
+typedef
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+(QUIC_API * QUIC_CONN_POOL_API_CLOSE)(
+    _In_ void* ConnPoolApi
+    );
+
+//
+// Creates a simple connection pool with NumberOfConnections connections
+// all with the same Context and Handler, and puts them in the
+// caller-supplied array.
+//
+// One of ServerName or ServerAddress *MUST* be supplied, and for the pool to
+// work correctly, the connections *MUST* be started with the same ServerName
+// and ServerPort as supplied in this call.
+//
+typedef
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Check_return_
+QUIC_STATUS
+(QUIC_API * QUIC_SIMPLE_CONN_POOL_CREATE_FN)(
```

Comment:
Let's call this just `QUIC_CONNECTION_POOL_CREATE_FN`, and the usage below `ConnectionPoolCreate`. I'd rather not include simple in the name. We can keep it wrapped in `QUIC_API_ENABLE_PREVIEW_FEATURES` to allow for possible changes in the future if we want to make it more complicated.